style: apply /style-guide pass to weave (#2739)
wandb/docs: weave/cookbooks/audio_with_weave.mdx

---
title: "Audio with Weave"
description: "Track and evaluate audio data in your LLM applications using W&B Weave's audio logging and tracing features."
keywords: ["OpenAI audio", "Realtime API", "GPT-4o audio", "audio tracing", "speech logging"]
---

<Note>
This is an interactive notebook. You can run it locally or use the following links:
- [Open in Google Colab](https://colab.research.google.com/github/wandb/docs/blob/main/weave/cookbooks/source/audio_with_weave.ipynb)
- [View source on GitHub](https://github.com/wandb/docs/blob/main/weave/cookbooks/source/audio_with_weave.ipynb)
</Note>

> This demo shows you how to use Weave to log and trace audio generated by OpenAI's audio APIs, so you can inspect prompts, audio outputs, and transcripts alongside the rest of your LLM application data. The demo targets developers who already build with OpenAI's audio models and want observability into their audio traces.

Reviewer: ```suggestion
This notebook shows you how to use Weave to log and trace audio generated by OpenAI's audio APIs, so you can inspect prompts, audio outputs, and transcripts alongside the rest of your LLM application data. The notebook targets developers who already build with OpenAI's audio models and want observability into their audio traces.
```

Author: Applied verbatim in 558ba397a.

> The demo first uses the OpenAI chat completions API with GPT 4o Audio Preview to generate audio responses to text prompts and track these in Weave.

Reviewer: ```suggestion
The notebook first uses the OpenAI chat completions API with GPT 4o Audio Preview to generate audio responses to text prompts and track these in Weave.
```

Author: Applied verbatim in 558ba397a.

<img src="/images/weave/cookbooks/openai-chat-completions-audio.png" alt="OpenAI chat completions API interface with GPT 4o Audio Preview integration and audio response generation workflow" />

> For the advanced use case, the demo uses the OpenAI Realtime API to stream audio in real time, so you can see how Weave captures both sides of a live conversation. Click the following thumbnail to view the [video demonstration](https://www.youtube.com/watch?v=lnnd73xDElw).

Reviewer: ```suggestion
For the advanced use case, the notebook uses the OpenAI Realtime API to stream audio in real time, so you can see how Weave captures both sides of a live conversation. Click the following thumbnail to view the [video demonstration](https://www.youtube.com/watch?v=lnnd73xDElw).
```

Author: Applied verbatim in 558ba397a.

## Audio streaming and storage example

With the dependencies installed and credentials loaded, you can now set up a call to OpenAI's completions endpoint with audio modality enabled. First, create the OpenAI client and initiate a Weave project so that Weave logs subsequent calls to your workspace.

```python lines
client = OpenAI(api_key=os.environ.get("OPENAI_API_KEY"))
weave.init("openai-audio-chat")
```

> Now, define the OpenAI completions request and add the Weave decorator (op). The `@weave.op()` decorator causes Weave to capture the function's inputs, outputs, and audio file in a trace.

Reviewer: ```suggestion
Now, define the OpenAI completions request and add the Weave decorator (op). The `@weave.op()` decorator instructs Weave to capture the function's inputs, outputs, and audio file in a trace.
```

Author: Applied verbatim in 558ba397a.

## Testing

> Now that the function is defined, run the following cell to invoke it end-to-end and confirm that audio is generated and logged. Weave stores the system and user prompts in a trace along with the output audio.

Reviewer: ```suggestion
After defining the function, run the following cell to invoke it end-to-end and confirm that audio is generated and logged. Weave stores the system and user prompts in a trace along with the output audio.
```

Author: Applied verbatim in 558ba397a.
